libs/core/src/main/java/de/westnordost/osmapi/common/GpxInputStreamWriter.java

```diff
@@ -0,0 +1,37 @@
+package de.westnordost.osmapi.common;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+
+import de.westnordost.osmapi.ApiRequestWriter;
+
+public class GpxInputStreamWriter implements ApiRequestWriter
+{
+	private static final String CHARSET = "UTF-8";
+
+	private InputStream gpx;
+
+	public GpxInputStreamWriter(InputStream in)
+	{
+		this.gpx = in;
+	}
+
+	@Override
+	public String getContentType()
+	{
+		return "text/plain";
+	}
+
+	public void write(OutputStream out) throws IOException
+	{
+		BufferedReader br = new BufferedReader(new InputStreamReader(gpx));
+		String line;
+		while ((line = br.readLine()) != null)
+		{
+			out.write((line + System.lineSeparator()).getBytes(CHARSET));
+		}
```

Comment:
A line within an XML file is somewhat arbitrary of a limit. Theoretically, depending on what software put together the XML, the whole file could be one line, in which case there would be no buffering at all.

It is also somewhat awkward to first separate by line and then put the lines back together.

So, how about this old-school? (Heh, actually shorter, too)

```suggestion
		byte[] buffer = new byte[8192];
		int length;
		while ((length = gpx.read(buffer )) != -1)
		{
			out.write(buffer , 0, length);
		}
```